### ISSUE TITLE
Invoke randomly fails with status code 3 on GitHub Actions CI with Python 3.7.12 on Ubuntu & MacOS

### ISSUE THREAD

**Issue by nialov** [pyinvoke/invoke#830]
# What I expected

`invoke` runs without failure as the nox command exits with status code 0. 

# What I got

Instead the invocation fails and returns exit code 3 which fails the workflow.

# Information

Relevant part of the log from [Github actions](https://github.com/nialov/drillcore-transformations/runs/4345943227?check_suite_focus=true) is below. 

~~~
Run invoke ci-test --python 3.7
  invoke ci-test --python 3.7
  shell: /bin/bash -e {0}
  env:
    pythonLocation: /Users/runner/hostedtoolcache/Python/3.7.12/x64
nox > Running session requirements
nox > Creating virtual environment (virtualenv) using python in .nox/requirements
nox > python -m pip install poetry
nox > poetry export --without-hashes --dev -o requirements.txt
nox > poetry export --without-hashes --dev -E docs -o docs_src/requirements.txt
nox > Session requirements was successful.
nox > Running session update_version
nox > Creating virtual environment (virtualenv) using python in .nox/update_version
nox > python -m pip install poetry-dynamic-versioning
nox > poetry-dynamic-versioning 
Version: 0.0.0.post1.dev0+337c7b7
Files with substitutions:
  - /Users/runner/work/drillcore-transformations/drillcore-transformations/drillcore_transformations/__init__.py
  - /Users/runner/work/drillcore-transformations/drillcore-transformations/CITATION.cff
nox > Session update_version was successful.
Error: Process completed with exit code 3.
~~~

Code for the task:

~~~
@task
def requirements(c):
    """
    Sync requirements.
    """
    c.run("nox --session requirements")

@task
def update_version(c):
    """
    Update pyproject.toml and package/__init__.py version strings.
    """
    c.run("nox --session update_version")


@task(pre=[requirements, update_version])
def ci_test(c, python=""):
    """
    Test suite for continous integration testing.

    Installs with pip, tests with pytest and checks coverage with coverage.
    """
    python_version = "" if len(python) == 0 else f"-p {python}"
    c.run(f"nox --session tests_pip {python_version}")

~~~

Code for the workflow yml:

~~~
jobs:
  pytest-coverage:
    name: Test and check coverage of 🐍 code
    strategy:
      matrix:
        python-version: [3.7, 3.8, 3.9]
        platform: [ubuntu-latest, macos-latest]
    runs-on: ${{ matrix.platform }}
    steps:
      - uses: actions/checkout@v2
      - name: Set up Python ${{ matrix.python-version }}
        uses: actions/setup-python@v2
        with:
          python-version: ${{ matrix.python-version }}
      - name: Install dependencies
        run: |
          python -m pip install --upgrade pip
          pip install nox invoke
      - name: Cache .nox
        uses: actions/cache@v2
        with:
          path: .nox
          key: ${{ runner.os }}-${{ matrix.python-version }}-nox-${{ hashFiles('noxfile.py') }}-${{ hashFiles('poetry.lock') }}-ci-test
      - name: Test with invoke -> nox -> pytest
        run: invoke ci-test --python ${{ matrix.python-version }}

~~~

# System specs

In short: Python 3.7.12 (Does not occur with 3.8 or 3.9), Ubuntu/MacOS.

Log is also saved as gist: https://gist.github.com/nialov/5b49e39999b0eabc57b2f15cb0df9c53
